### PR TITLE
Stakes and votes getvoteaccounts

### PR DESCRIPTION
**Pull request by musitdev**
Implementation  of : 
 * getVoteAccounts.
 * getSlotLeaders: search in the 2 cached current epochs.

These call has been tested in local using some stress rpc call.

They need to be tested by comparing the result with a real validator to validate that the values are correct.

### CODE REVIEW

File: cluster-endpoints/src/rpc_polling/poll_slots.rs [hunk 1]
```diff
@@ -5,7 +5,7 @@ use solana_client::nonblocking::rpc_client::RpcClient;
 use solana_lite_rpc_core::{structures::slot_notification::SlotNotification, AnyhowJoinHandle};
 use solana_sdk::{commitment_config::CommitmentConfig, slot_history::Slot};
 use tokio::sync::broadcast::Sender;
-const AVERAGE_SLOT_CHANGE_TIME: Duration = Duration::from_millis(400);
+const AVERAGE_SLOT_CHANGE_TIME: Duration = Duration::from_millis(800);
```
Comment (godmodegalactus) [inline cluster-endpoints/src/rpc_polling/poll_slots.rs:8]: Why 800 should be around 400 right.

File: config.example.json [hunk 2]
```diff
@@ -10,6 +10,7 @@
   "transaction_retry_after_secs": 3,
   "quic_proxy_addr": null,
   "use_grpc": false,
+  "activate_leader_schedule": false,
```
Comment (godmodegalactus) [inline config.example.json:13]: Should be `calculate_leader_schedule_form_geyser`

File: core/src/structures/leaderschedule.rs [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+}
+
+impl CalculatedSchedule {
+    pub async fn get_leader_schedule_for_slot(
+        &self,
+        slot: Option<u64>,
+        commitment: Option<CommitmentConfig>,
+        data_cache: &DataCache,
+    ) -> Option<HashMap<String, Vec<usize>>> {
+        log::info!(
```
Comment (godmodegalactus) [inline core/src/structures/leaderschedule.rs:51]: debug

File: lite-rpc/src/lib.rs [hunk 4]
```diff
@@ -33,7 +33,7 @@ pub const DEFAULT_TRANSACTION_CONFIRMATION_STATUS: TransactionConfirmationStatus
     TransactionConfirmationStatus::Finalized;
 
 #[from_env]
-pub const DEFAULT_GRPC_ADDR: &str = "http://127.0.0.0:10000";
+pub const DEFAULT_GRPC_ADDR: &str = "http://localhost:10000";
```
Comment (godmodegalactus) [inline lite-rpc/src/lib.rs:36]: Should be 127.0.0.1 i think there was an issue before if we specify localhost for websockets or grpc, don't remember. But if you confirm it is fixed now then you can keep it.

File: stake_vote/src/bootstrap.rs [hunk 5]
```diff
@@ -0,0 +1,404 @@
...
+        .ok()
+}
+
+/*
+Bootstrap state changes
+
+  InitBootstrap
+       |
+  |Fetch accounts|
+
```
Comment (godmodegalactus) [inline stake_vote/src/bootstrap.rs:68]: How do you create these trees ?

File: stake_vote/src/lib.rs [hunk 6]
```diff
@@ -0,0 +1,364 @@
...
+    mut slot_notification: SlotStream,
+    mut vote_account_rpc_request: Receiver<(
+        GetVoteAccountsConfig,
+        tokio::sync::oneshot::Sender<RpcVoteAccountStatus>,
+    )>,
+    rpc_client: Arc<RpcClient>,
+    grpc_url: String,
+) -> anyhow::Result<tokio::task::JoinHandle<()>> {
+    log::info!("Start Stake and Vote loop on :{grpc_url}.");
+    let mut stake_vote_geyzer_stream = subscribe_geyzer_stake_vote_owner(grpc_url.clone()).await?;
```
Comment (godmodegalactus) [inline stake_vote/src/lib.rs:51]: geyser*

File: stake_vote/src/lib.rs [hunk 7]
```diff
@@ -0,0 +1,364 @@
...
+    let jh = tokio::spawn(async move {
+        //Stake account management struct
+        let mut stakestore = stake::StakeStore::new(STAKESTORE_INITIAL_CAPACITY);
+
+        //Vote account management struct
+        let mut votestore = vote::VoteStore::new(VOTESTORE_INITIAL_CAPACITY);
+
+        //Init bootstrap process
+        let mut current_schedule_epoch =
+            crate::bootstrap::bootstrap_scheduleepoch_data(&data_cache).await;
```
Comment (godmodegalactus) [inline stake_vote/src/lib.rs:63]: bootstrap_schedule_epoch_data

**Review by godmodegalactus — CHANGES_REQUESTED** [review]

File: cluster-endpoints/src/rpc_polling/poll_slots.rs (hunk 1 above)
Comment (musitdev) [reply cluster-endpoints/src/rpc_polling/poll_slots.rs]: Corrected

File: config.example.json (hunk 2 above)
Comment (musitdev) [reply config.example.json]: Done

File: core/src/structures/leaderschedule.rs (hunk 3 above)
Comment (musitdev) [reply core/src/structures/leaderschedule.rs]: Done

File: lite-rpc/src/lib.rs (hunk 4 above)
Comment (musitdev) [reply lite-rpc/src/lib.rs:36]: The issue was with 127.0.0.1. For example on amd81 server it doesn't work. That why I change to localhost.

File: stake_vote/src/bootstrap.rs (hunk 5 above)
Comment (musitdev) [reply stake_vote/src/bootstrap.rs:68]: I use pipe key (Alt Gr 6) on french keyboard.

File: stake_vote/src/lib.rs (hunk 6 above)
Comment (musitdev) [reply stake_vote/src/lib.rs]: Corrected. I change all spelling geyzer.

File: stake_vote/src/lib.rs (hunk 7 above)
Comment (musitdev) [reply stake_vote/src/lib.rs]: Changed

File: stake_vote/src/bootstrap.rs (hunk 5 above)
Comment (godmodegalactus) [reply stake_vote/src/bootstrap.rs:68]: I though you use some library and not make it by hand.

**Review by godmodegalactus — APPROVED** [review]

**Pull request merged**